core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingResultConverter.java

```diff
@@ -112,7 +125,8 @@ public static List<PathWaypointResult> getWaypointsOnRoute(
 
                         if (trackRangeOffset >= 0 && trackRangeOffset <= truncatedRange.getLength()) {
                             var location = truncatedRange.offsetLocation(trackRangeOffset);
-                            waypoints.add(PathWaypointResult.userDefined(location, trackRangeOffset));
+                            var pathOffset = routeOffset + offset + trackRangeOffset;
```

Comment:
same as above, I think this only works when the path goes in the same direction as the track section